Update backend to use the COOL Terraform account

jsf9k opened the pull request:
## 🗣 Description

This pull request updates the backend configuration to use the COOL Terraform account.

## 💭 Motivation and Context

We should be using the COOL Terraform account for all the COOL deployments.

## 🧪 Testing

I made sure that all the pre-commit hooks passed.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

jsf9k committed:
Update the backend config to use the COOL Terraform account

jsf9k committed:
Rename a variable

The new name better explains what the variable contains.

jsf9k committed:
Tweak language

felddy commented on backend.tf:
```diff
@@ -1,9 +1,9 @@
+    bucket         = "cisa-cool-terraform-state"
     dynamodb_table = "terraform-state-lock"
-    profile        = "playground"
+    role_arn       = "arn:aws:iam::608004238745:role/TerraformStateAccess"
```
I can't remember what we decided.  Weren't we keeping all the ARNs inside our aws configs, and referencing the role names?  (See the "roles" tab of the "CISA COOL AWS Accounts" spreadsheet.  

This allows us to use roles without checking in our account numbers to version control.

jsf9k replied:
> I can't remember what we decided. Weren't we keeping all the ARNs inside our aws configs, and referencing the role names? (See the "roles" tab of the "CISA COOL AWS Accounts" spreadsheet.
> 
> This allows us to use roles without checking in our account numbers to version control.

I had it pointing to profiles before, but I thought we decided to use ARNs instead.  See, for example, [this commit](https://github.com/cisagov/pca-terraform/pull/7/commits/2a0294214e5b921330c8b4afe7a85ea6968d64ea).

@dav3r may be able to shed some light on this.  His memory tends to be better than mine.

jsf9k replied:
Using ARNs means one less layer of obfuscation, and one can look at the code in the repo and see what is going on without any knowledge of the way that we set up our boto3 profiles.

I think what makes the most sense is to use a profile to configure the backend (since variables are not allowed in that configuration block and we don't want actual ARN values out in the open) but use ARNs specified in variables (and hence in a `*.tfvars` file that is not committed) to configure the providers.  This may be what @felddy meant all along, and I simply misunderstood his intent.

See commit 622b7e0.

felddy reviewed (approved):
Approved.  See my comment about profiles vs. ARNs.

jsf9k committed:
Use a profile instead of a role ARN to configure the backend

Putting account IDs and such in a public repo is not a big deal, but
some folks are already squeamish about using AWS so we may as well not
give them anything extra to worry about.

mcdonnnj reviewed (approved):
Makes sense to me. 👍
